src/libtest/test_result.rs

```diff
@@ -49,9 +51,19 @@ pub fn calc_result<'a>(
                 if desc.allow_fail {
```

Comment:
Could be simplified to `else if desc.allow_fail` and `else if let Some(panic_str) = maybe_panic_str`

Reply:
Thanks, looks much better and will look nicer once allow_fail is removed too